metabase/metabase
# Fix loading locale in setup

Fixes https://github.com/metabase/metabase/issues/21403
Fixes https://github.com/metabase/metabase/issues/21404

Changes:
- Adds types for setup state
- Reloads locale in setup when it changes
- Gracefully handles locale loading issues

How to test:
- Go to browser settings and make your preferred language not English
- Run a fresh Metabase instance
- Make sure that the initial page is translated to your browser language
- Go to the language selection step and change languages
- Make sure that all steps are translated when the language changes

Please note that some translations for setup are broken or missing in `.po` files.
